### PR TITLE
feat: Sort picks by the logged user

### PR DESCRIPTION
**Pull request by kevinszuchet**
Related to https://github.com/decentraland/marketplace/issues/1567

### CODE REVIEW

File: src/ports/picks/component.ts [hunk 1]
```diff
@@ -37,19 +37,26 @@ export function createPicksComponent(components: Pick<AppComponents, 'pg'>): IPi
...
-          }
-        ) AS temp
-        ORDER BY user_address
-        LIMIT ${limit} OFFSET ${offset}
-    `)
+    const { limit, offset, power, userAddress } = options
+    const query = SQL`SELECT user_address, COUNT(*) OVER() as picks_count`
+
+    if (userAddress) {
+      query.append(SQL`, (CASE WHEN user_address = ${userAddress} THEN 1 ELSE 0 END)::BOOLEAN AS picked_by_user`)
```
Comment (LautaroPetaccio) [inline src/ports/picks/component.ts:44]: What do you think about omitting the boolean here as the number will work as well for the order?
```suggestion
      query.append(SQL`, (CASE WHEN user_address = ${userAddress} THEN 1 ELSE 0 END) AS picked_by_user`)
```

File: src/ports/picks/component.ts [hunk 2]
```diff
@@ -37,19 +37,26 @@ export function createPicksComponent(components: Pick<AppComponents, 'pg'>): IPi
...
+      query.append(SQL`, false AS picked_by_user`)
+    }
+
+    query.append(
+      SQL` FROM (
+        SELECT DISTINCT favorites.picks.user_address FROM favorites.picks, favorites.voting
+        WHERE favorites.picks.item_id = ${itemId}
+        AND favorites.voting.user_address = favorites.picks.user_address AND favorites.voting.power >= ${power ?? DEFAULT_VOTING_POWER}
+      ) AS temp
+      ORDER BY picked_by_user DESC, user_address
```
Comment (LautaroPetaccio) [inline src/ports/picks/component.ts:55]: What if we change the default ordering from `user_address` to created_at here as well? We would return the listing ordered by its creation date, which seems a bit better:
```suggestion
      ORDER BY picked_by_user DESC, created_at DESC
```

File: src/ports/picks/component.ts [hunk 3]
```diff
@@ -41,18 +41,18 @@ export function createPicksComponent(components: Pick<AppComponents, 'pg'>): IPi
...
-      query.append(SQL`, (CASE WHEN user_address = ${userAddress} THEN 1 ELSE 0 END)::BOOLEAN AS picked_by_user`)
+      query.append(SQL`, user_address = ${userAddress} AS picked_by_user`)
     } else {
       query.append(SQL`, false AS picked_by_user`)
     }
 
     query.append(
       SQL` FROM (
-        SELECT DISTINCT favorites.picks.user_address FROM favorites.picks, favorites.voting
+        SELECT DISTINCT favorites.picks.user_address, favorites.picks.created_at FROM favorites.picks, favorites.voting
```
Comment (LautaroPetaccio) [inline src/ports/picks/component.ts:51]: Won't this change return more than one entry for a `user_address`? That is, before, we were getting the unique results for a `user_address`, but with the `created_at` we're going to be getting more than one as the tuple (`favorites.picks.user_address, favorites.picks.created_at`) is the one to check for uniqueness. `SELECT DISTINCT ON` over the `favorites.picks.user_address` column might fix the issue.

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/ports/picks/component.ts (hunk 3 above)
Comment (kevinszuchet) [reply src/ports/picks/component.ts]: Great catch! Thanks for the suggestion.

**Review by LautaroPetaccio — APPROVED** [review]
Great change 🚀

**Pull request merged**